File: Languages/Python/Day016.py
# Languages - Python - Day 16
# Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

# Create a function that checks if a number is an Armstrong Number. An Armstrong Number is a number that is equal to the sum of its digits each raised to the power of the number of digits.

# function to check if a number is an Armstrong Number
def is_armstrong_number(n):
    if not isinstance(n, int) or n < 0:
        print("Please enter a positive integer.")
        return False

    num_str = str(n)
    num_digits = len(num_str)
    total = sum(int(digit) ** num_digits for digit in num_str)

    return total == n

# function to prompt the user for input
def prompt_user():
    try:
        user_input = input(
            "Enter a positive integer to check if it's an Armstrong Number: "
        )
        num = int(user_input)

        if num < 0:
            print("Invalid input. Please enter a positive integer.")
            return

        if is_armstrong_number(num):
            print(f"{num} is an Armstrong Number.")
        else:
            print(f"{num} is not an Armstrong Number.")

    except ValueError:
        print("Invalid input. Please enter a positive integer.")


# Run the prompt
prompt_user()

File: Languages/Kotlin/Day016.kt
// Languages - Kotlin - Day 16
// Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

// Create a function that checks if a number is an Armstrong Number. An Armstrong Number is a number that is equal to the sum of its digits each raised to the power of the number of digits.

import java.util.Scanner
import kotlin.math.pow

// Function to check if a number is an Armstrong Number
fun isArmstrongNumber(num: Int): Boolean {
    if (num < 0) {
        println("Please enter a positive integer.")
        return false
    }

    val numStr = num.toString()
    val numDigits = numStr.length
    var sum = 0

    for (digitChar in numStr) {
        val digit = digitChar.toString().toInt()
        sum += digit.toDouble().pow(numDigits).toInt()
    }

    return sum == num
}

// Function to prompt user input
fun promptUser() {
    val scanner = Scanner(System.`in`)
    print("Enter a positive integer to check if it's an Armstrong Number: ")

    if (scanner.hasNextInt()) {
        val num = scanner.nextInt()
        if (num < 0) {
            println("Invalid input. Please enter a positive integer.")
        } else {
            if (isArmstrongNumber(num)) {
                println("$num is an Armstrong Number.")
            } else {
                println("$num is not an Armstrong Number.")
            }
        }
    } else {
        println("Invalid input. Please enter a positive integer.")
    }
}

// Main function to run the program
fun main() {
    promptUser()
}
